src/Expanders.jl

```diff
@@ -499,6 +507,36 @@ function Selectors.runner(::Type{REPLBlocks}, x, page, doc)
     page.mapping[x] = Base.Markdown.Code("julia", rstrip(takebuf_string(out)))
 end
 
+# @setup
+# --------
+
+function Selectors.runner(::Type{SetupBlocks}, x, page, doc)
+    matched = Utilities.nullmatch(r"^@setup[ ](.+)$", x.language)
+    isnull(matched) && error("invalid '@setup <name>' syntax: $(x.language)")
+    # The sandboxed module -- either a new one or a cached one from this page.
+    name = Utilities.getmatch(matched, 1)
+    sym  = isempty(name) ? gensym("ex-") : Symbol("ex-", name)
+    mod  = get!(page.globals.meta, sym, Module(sym))::Module
+    # Evaluate the code block. We redirect STDOUT/STDERR to `buffer`.
+    result, buffer = nothing, IOBuffer()
+    for (ex, str) in Utilities.parseblock(x.code, doc, page)
+        (value, success, backtrace, text) = Utilities.withoutput() do
+            cd(dirname(page.build)) do
+                eval(mod, :(ans = $(eval(mod, ex))))
+            end
+        end
+        result = value
+        print(buffer, text)
+        if !success
+            Utilities.warn(page.source, "failed to run code block.\n\n$(value)")
+            page.mapping[x] = x
+            return
+        end
+    end
```

Comment:
Since a `@setup` block doesn't need to do the parsing piece-by-piece it would probably be better to just do

```julia
eval(mod, :(include_string($(x.code))))
```